From: Alexander Block <ablock84@googlemail.com>
Date: Sat, 28 Jul 2012 14:13:35 +0200
Patch-mainline: 3.7
Git-commit: 17589bd96eeb7c2ef2d3baeb05005a24932cd1e9
References: FATE#312888
Subject: [PATCH] Btrfs: fix memory leak for name_cache in
 send/receive

When everything is done, name_cache_free is called which however
forgot to call kfree on the cache entries.

Signed-off-by: Alexander Block <ablock84@googlemail.com>
Signed-off-by: David Sterba <dsterba@suse.cz>
---
 fs/btrfs/send.c |    1 +
 1 file changed, 1 insertion(+)

--- a/fs/btrfs/send.c
+++ b/fs/btrfs/send.c
@@ -1822,6 +1822,7 @@ static void name_cache_free(struct send_
 
 	list_for_each_entry_safe(nce, tmp, &sctx->name_cache_list, list) {
 		name_cache_delete(sctx, nce);
+		kfree(nce);
 	}
 }
 
